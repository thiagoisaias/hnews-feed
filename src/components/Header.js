import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import Search from "./Search";
import FiltersBar from "./FiltersBar";
import LogoImagePath from "../images/logo-hn.png";


const Container = styled.div`
  width: 100%;
  max-width: 900px;
  position: fixed;
  top: 0;
`;


const TopInnerContainer = styled.div`
  height: 80px;
  display: flex;
  align-items: center;
  background-color: #333;
`;

const Title = styled.div`
  color: #fff;
  font-size: 1.25em;
  cursor: default;
`;

const LogoContainer = styled.div`
  display: flex;
  align-items: center;
  float: left;
  margin: 0 32px;
  @media (max-width: 660px) {
    display: none;
  }
`;

const LogoLink = styled.a`
  line-height: 0;
  &:hover {
    opacity: 1;
  }
`;

const LogoImage = styled.img`
  width: 40px;
  height: 40px;
  margin-right: 16px;
`;

const Header = props => {
  return (
    <Container>
      <TopInnerContainer>
        <LogoContainer>
          <LogoLink href="/">
            <LogoImage src={LogoImagePath} />
          </LogoLink>
          <Title>Hacker News</Title>
        </LogoContainer>
        <Search {...props} />
      </TopInnerContainer>
      <FiltersBar
        sortByDate={props.sortByDate}
        sortByPopularity={props.sortByPopularity}
      />
    </Container>
  );
};

Header.propTypes = {
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  sortByDate: PropTypes.func.isRequired,
  sortByPopularity: PropTypes.func.isRequired
};

export default Header;
