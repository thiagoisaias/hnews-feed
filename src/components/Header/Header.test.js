import React from "react";
import Enzyme, { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import Header from "./Header";

describe("Header", () => {
  it("renders properly", () => {
    const props = {
      value: "",
      onChange: jest.fn(),
      onSubmit: jest.fn(),
      sortByDate: jest.fn(),
      sortByPopularity: jest.fn()
    };
    const HeaderTest = shallow(<Header {...props} />);
    expect(shallowToJson(HeaderTest)).toMatchSnapshot();
  });
});
