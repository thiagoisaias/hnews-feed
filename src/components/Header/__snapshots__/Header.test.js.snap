// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders properly 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      <styled.a
        href="/"
      >
        <styled.img
          src="logo-hn.png"
        />
      </styled.a>
      <styled.div>
        Hacker News
      </styled.div>
    </styled.div>
    <Search
      onChange={[Function]}
      onSubmit={[Function]}
      sortByDate={[Function]}
      sortByPopularity={[Function]}
      value=""
    />
  </styled.div>
  <FiltersBar
    sortByDate={[Function]}
    sortByPopularity={[Function]}
  />
</styled.div>
`;
